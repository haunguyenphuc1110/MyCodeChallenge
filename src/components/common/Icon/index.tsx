import React from 'react'
import { View, StyleSheet } from 'react-native'
import FontAwesomeIcons from 'react-native-vector-icons/FontAwesome5'
import MaterialIcons from 'react-native-vector-icons/MaterialIcons'
import Ionicons from 'react-native-vector-icons/Ionicons'
import AntDesignIcons from 'react-native-vector-icons/AntDesign'

import { COLORS } from 'Constants'

interface IconProps {
  type?: string
  name: string
  style?: object
  color?: string
  size?: number
  hasBackground?: boolean
  onPress?: () => void
}

const mappingIcon = {
  fontawesome: FontAwesomeIcons,
  material: MaterialIcons,
  ion: Ionicons,
  antdesign: AntDesignIcons
} as any

const RNIcon: React.FC<IconProps> = ({
  type = 'fontawesome',
  name,
  style,
  color = COLORS.white,
  size = 12,
  hasBackground = false,
  onPress
}) => {
  const Icon = mappingIcon[type]

  if (hasBackground) {
    return (
      <View style={[styles.wrapper, style]}>
        <Icon
          name={name}
          color={color}
          size={size}
          onPress={onPress} />
      </View>
    )
  }

  return (
    <Icon
      name={name}
      style={style}
      color={color}
      size={size}
      onPress={onPress} />
  )
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: COLORS.transparent,
    padding: 5,
    borderRadius: 50
  }
})

export default RNIcon
