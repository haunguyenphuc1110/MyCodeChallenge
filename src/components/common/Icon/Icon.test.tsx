import React from 'react'
import renderer from 'react-test-renderer'

import Icon from './index'

describe('Icon', () => {
  describe('snapshot render', () => {
    test('with initial props', () => {
      const tree = renderer.create(<Icon name={'award'} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
    test('with hasBackground props is true', () => {
      const tree = renderer.create(<Icon name={'award'} hasBackground={true} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
