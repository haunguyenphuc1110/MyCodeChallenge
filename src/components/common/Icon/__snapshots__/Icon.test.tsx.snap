// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon snapshot render with hasBackground props is true 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 50,
        "padding": 5,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "900",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon snapshot render with initial props 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome5Free-Solid",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "900",
      },
    ]
  }
>
  
</Text>
`;
